crates/pipeline_manager/src/db/mod.rs

```diff
@@ -261,15 +262,170 @@ pub(crate) struct ProgramDescr {
+/// then polls the `GET /pipeline` endpoint to monitor the actual status
+/// of the pipeline until its `state.current_status` attribute changes
+/// to "paused" indicating that the pipeline has been successfully
+/// initialized, or "failed", indicating an error.
+#[derive(Deserialize, Serialize, ToSchema, Eq, PartialEq, Debug, Clone, Copy)]
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
-pub(crate) enum PipelineStatus {
+pub enum PipelineStatus {
+    /// Pipeline has not been started or has been shut down.
+    ///
+    /// The pipeline remains in this state until the user triggers
+    /// a deployment by invoking the `/deploy` endpoint.
     Shutdown,
-    Deployed,
-    Running,
+
+    /// The runner triggered a deployment of the pipeline and is
+    /// waiting for the pipeline HTTP server to come up.
+    ///
+    /// In this state, the runner provisions a runtime for the pipeline
+    /// (e.g., a Kubernetes pod or a local process), starts the pipeline
+    /// within this runtime and waits for it to start accepting HTTP
+    /// requests.
+    ///
+    /// The user is unable to communicate with the pipeline during this
+    /// time.  The pipeline remains in this state until:
+    ///
+    /// 1. Its HTTP server is up and running; the pipeline transitions
+    ///    to the [`Initializing`](`Self::Initializing`) state.
+    /// 2. A pre-defined timeout has passed.  The runner performs forced
+    ///    shutdown of the pipeline; returns to the
+    ///    [`Shutdown`](`Self::Shutdown`) state.
+    /// 3. The user cancels the pipeline by invoking the `/shutdown` endpoint.
+    ///    The manager performs forced shutdown of the pipeline, returns to
+    ///    the [`Shutdown`](`Self::Shutdown`) state.
+    Provisioning,
+
+    /// The pipeline is initializing its internal state and connectors.
+    ///
+    /// This state is part of the pipeline's deployment process.  In this state,
+    /// the pipeline's HTTP server is up and running, but its query engine
+    /// and input and output connectors are still initializing.
+    ///
+    /// The pipeline remains in this state until:
+    ///
+    /// 1. Intialization completes successfully; the pipeline transitions
+    ///    to the [`Paused`](`Self::Paused`) state.
+    /// 2. Intialization fails; transitions to the [`Failed`](`Self::Failed`) state.
+    /// 3. A pre-defined timeout has passed.  The runner performs forced shutdown of
+    ///    the pipeline; returns to the [`Shutdown`](`Self::Shutdown`) state.
+    /// 4. The user cancels the pipeline by invoking the `/shutdown` endpoint.
+    ///    The manager performs forced shutdown of the pipeline, returns to the
+    ///    [`Shutdown`](`Self::Shutdown`) state.
+    Initializing,
+
+    /// The pipeline is fully initialized, but data processing has been paused.
+    ///
+    /// The pipeline remains in this state until:
+    ///
+    /// 1. The user starts the pipeline by invoking the `/start` endpoint.
```

Comment:
I'd avoid bringing up the user here, and talking strictly about what the pipeline being in this state means. Going forward, these workflows could even be triggered in the background by a cluster manager.

Reply:
This documentation propagates to OpenAPI docs, so it makes sense to focus on the user perspective here, even if there will be other types of workflows in the future.